### ISSUE TITLE
v1.3.0 no default constructor for LockGuard when compiling without CEREAL_THREAD_SAFE set

### ISSUE THREAD

**Issue by rsjaffe** [USCiLab/cereal#592]
A project that compiled fine with 1.2.2 failed with 1.3.0 with the message: 
```
C2512	'cereal::detail::StaticObject<cereal::detail::Versions>::LockGuard': no appropriate default constructor available	MIDI2LR_App	C:\Users\rsjaf\source\repos\MIDI2LR\cereal\details\static_object.hpp	115		
```
(MSVC 16.3.7). My project is at https://github.com/rsjaffe/MIDI2LR

Looking at the code in static_object.hpp, I see that the problem occurs with:

```
        static LockGuard lock()
        {
          #if CEREAL_THREAD_SAFE
          static std::mutex instanceMutex;
          return LockGuard{instanceMutex};
          #else
          return LockGuard{};
          #endif
        }
```
as the definition of Lockguard doesn't have a no-argument constructor since there is a user-defined constructor.

**Comment by AzothAmmo** [comment]
Guess we need CI running on MSVC 2019! Can you try providing a defaulted constructor for `StaticObject` and see if that makes things compile? I just made a branch to try and get the CI including 2019 passing with this change.

**Comment by rsjaffe** [comment]
Adding `LockGuard() = default;` to the public section of class LockGuard fixes it. Removing `LockGuard(LockGuard const &) = default;` does not produce any additional warnings, even when "all rules" for code analysis is enabled in MSVC.

**Comment by AzothAmmo** [comment]
Should be fixed by #593
